### PR TITLE
Add files needed for users log in and log out

### PR DESCRIPTION
**Pull request by sebastiancaraballo**
Trello board reference:
https://trello.com/c/YHXtrAp8

Description:
Add all files needed for users log in and log out.

### CODE REVIEW

**Review by mdornel — APPROVED** [review]
Good Work!

File: spec/requests/api/v1/sessions/destroy_spec.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+describe 'DELETE api/v1/users/sign_out', type: :request do
+  let(:user) { create(:user) }
+
+  it 'returns a succesful response' do
+    delete destroy_user_session_path, headers: auth_headers, as: :json
+    expect(response).to have_http_status(:success)
+  end
+
+  it 'can not log out twice' do
```
Comment (gonzaM) [inline spec/requests/api/v1/sessions/destroy_spec.rb]: I'd prefer adding a context `'when the user is not logged in'` then you can test that the response is not found. I don't like to call twice the endpoint.

**Review by gonzaM — COMMENTED** [review]
Remember to update documentation

**Review by mdornel — COMMENTED** [review]
Remember to update the docs

File: apiary.apib [hunk 2]
```diff
@@ -42,14 +39,84 @@ For GENDER attribute, possible values are: __'male'__, __'female'__ and __'other
...
+    + Body
+            {
+              "success": true
+            }
+
++ Response 404
+    + Body
+            {
+              "error": "User was not found or was not logged in."
+            }
```
Comment (gonzaM) [inline apiary.apib]: Add empty line

**Review by gonzaM — APPROVED** [review]

File: spec/requests/api/v1/sessions/create_spec.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  let(:password) { 'password' }
+  let(:token) do
+    {
+      '70crCAAYmNP1xLkKKM09zA' =>
+      {
+        'token' => '$2a$10$mSeRnpVMaaegCpn3AhORGe5wajFhgMoBjGIrMwq4Qq2mP6f/OHu1y',
+        'expiry' => 153_574_356_4
+      }
+    }
+  end
```
Comment (gabrielbursztein2) [inline spec/requests/api/v1/sessions/create_spec.rb]: Are you using this token? 🤔

**Review by gabrielbursztein2 — APPROVED** [review]

File: spec/requests/api/v1/sessions/create_spec.rb (hunk 3 above)
Comment (gabrielbursztein2) [inline spec/requests/api/v1/sessions/create_spec.rb]: Try removing this and the tokens from the line below and check if the sign in works. I don't think we need this.

**Pull request merged**